gnome-initial-setup/gnome-initial-setup-copy-worker.c

```diff
@@ -69,18 +71,47 @@ move_file_from_homedir (GFile       *src_base,
 static void
 unlock_keyring (const gchar *initial_setup_homedir)
 {
-  gchar *file;
-  gchar *password = NULL;
+  g_autofree gchar *file = g_build_filename (initial_setup_homedir, ".config/password", NULL);
+  gsize password_size;
+  g_autofree gchar *password = NULL;
+  SecretValue *secret = NULL;
+  SecretService *service = NULL;
+  g_autoptr(GError) error = NULL;
+
+  if (!g_file_get_contents (file, &password, &password_size, &error)) {
+    g_warning ("Unable to read user password file: %s", error->message);
+    goto out;
+  }
 
-  file = g_build_filename (initial_setup_homedir, ".config/password", NULL);
-  if (g_file_get_contents (file, &password, NULL, NULL))
-    gnome_keyring_unlock_sync ("login", password);
-  else
-    g_warning ("Unable to read user password file");
+  service = secret_service_get_sync (SECRET_SERVICE_OPEN_SESSION, NULL, &error);
```

Comment:
Is it OK to make blocking D-Bus calls here? Is this function going to block the UI?

(I have no idea, but thought I should raise it anyway.)

Reply:
I realise that if this is a problem, the refactoring will be non-trivial and can probably be deferred.

Reply:
No. This is a stand-alone program that runs (conditionally) at login. See `/usr/etc/xdg/autostart/gnome-initial-setup-copy-worker.desktop`.

There are reasons to feel sad about the way this works but I don't think this particular point is a problem. (Don't worry, though: the UI also makes blocking D-Bus calls constantly.)